src/TrainDetectionSystem/TrainDetectionSystem.cs

```diff
@@ -33,38 +34,47 @@ public class TrainDetectionSystem : BackgroundService
         AsyncDuplexStreamingCall<SciPacket, SciPacket> _currentConnection;
         private Dictionary<string, TvpsOccupancyStatus> _trackSectionStatuses;
 
-        public TrainDetectionSystem(ILogger<TrainDetectionSystem> logger, IConfiguration configuration)
+        private TDSState _tdsState;
+        public TDSState TdsState { get { return _tdsState; } }

```

Comment:
Nit: you could make this a readonly property (initializable from the constructor only) and save the private variable

`public TDSState TdsState { get; }`

Reply:
solved